src/pyhf/infer/calculators.py

```diff
@@ -252,6 +286,22 @@ def __init__(
         self.calc_base_dist = calc_base_dist
         self.sqrtqmuA_v = None
 
+    @property
+    def fitted_pars(self):
+        """
+        Best-fit parameters of the fits that have been run in
+        :py:meth:`self.teststatistic <pyhf.infer.calculators.AsymptoticCalculator.teststatistic>`.
+
+        Returns:
+            ~pyhf.infer.calculators.HypoTestFitResults:  The collection of fitted parameter tensors.
+        """
+        try:
+            return self._fitted_pars
+        except AttributeError:
+            if self.sqrtqmuA_v is None:
+                raise RuntimeError("need to call .teststatistic(poi_test) first")
+            raise
+
```

Comment:
This is not a needed function. Just rename `self._fitted_pars` to `self.fitted_pars` and it's accessible there, and harmonizes with `self.sqrtqmuA_v`.

Reply:
I am a bit worried that without an error message that clearly says "need to call .teststatistic(poi_test) first" (like the other functions in the calculator that depend on it having been called) it could be very confusing for people not familiar with how the class works. But I see the point about it harmonizing, so now it is initialised to `None`, like `self.sqrtqmuA_v`

Reply:
yeah seems like a good use for `@property` actually. maybe `sqrtqmuA_v` shold be guarded ilke tthat as well?